File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        # length = len(nums)

        # for i in range(length-1):
        #     for j in range(i+1,length):
        #         candidate = nums[i]+nums[j]

        #         if candidate == target:
        #             return [i,j]
        
        # return None
        

        dit = {}

        for i, num in enumerate(nums):
            candidate = target - nums[i]

            if candidate in dit:
                return [dit[candidate],i]
            else:
                dit[num] = i
        
        return None

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        
//         for(int i=0; i<nums.length-1; i++) {
//             for(int j=i+1; j<nums.length; j++) {
//                 int candidate = nums[i] + nums[j];
                
//                 if(candidate ==target) {
//                     return new int [] {i, j};
//                 }
//             }
//         }
//         return null;
        
        Map<Integer,Integer> dit = new HashMap<>();
        
        for(int i=0; i< nums.length;i++) {
            
            int candidate = target - nums[i];
            
            if(dit.containsKey(candidate)) {
                return new int [] {dit.get(candidate), i};
            } else {
                dit.put(nums[i], i);
            }
        }
        return null;
        
    }
}